fairlearn/datasets/_fetch_data.py

```diff
@@ -0,0 +1,91 @@
+import warnings
+from ._common import FairnessWarning
+
+from sklearn.datasets import fetch_openml
+
+
+def fetch_boston(*, version='1', cache=True, data_home=None,
+                 as_frame=False, return_X_y=False, warn=True):
+    """Load the boston housing dataset (regression).
+
+    Download it if necessary.
+
+    ==============   ==============
+    Samples total               506
+    Dimensionality               13
+    Features                   real
+    Target           real 5. - 50.
+    ==============   ==============
+
+    This dataset has known fairness issues. There's a "lower status of
+    population" (LSTAT) parameter that you need to look out for and a column
+    that is a derived from the proportion of people with a black skin color
+    that live in a neighborhood (B). This dataset should be used to explore
+    and benchmark fairness. Predicting a house-price while ignoring these
+    sensitive attributes is naive.
+
+    Parameters
+    ----------
+    version : integer or 'active', default='active'
+        Version of the dataset. Can only be provided if also ``name`` is given.
+        If 'active' the oldest version that's still active is used. Since
+        there may be more than one active version of a dataset, and those
+        versions may fundamentally be different from one another, setting an
+        exact version is highly recommended.
+
+    cache : boolean, default=True
+        Whether to cache downloaded datasets using joblib.
+
+    data_home : optional, default: None
+        Specify another download and cache folder for the datasets. By default
+        all scikit-learn data is stored in '~/scikit_learn_data' subfolders.
```

Comment:
maybe use a `~/.fairlearn` folder or something instead?

Reply:
There's something to be said for that. Curious to hear if anybody has an objection to this.

Reply:
I agree with @adrinjalali . It makes more sense to store data in a folder with "fairlearn" in its name. I don't have a strong opinion if it should be named ".fairlearn" or "fairlearn-data" or something else of the sort.

Reply:
I’m happy with either one. Just make sure this is a directory that one usually has permission. I remember running into trouble with shap’s dataset utility on CircleCi because it downloaded the data into the directory where the package is stored. The CI didn’t have permissions for that so it failed. There’s a special case for it under examples/quickstart right now. Would love to never have to do this again.

Reply:
I don't know about windows systems actually. But for now I've implemented it such that we save it in;

```python
pathlib.Path().home() / ".fairlearn-data"
```

I think the home folder is allocated to the user, so this should suffice.